server/memphis_handlers_functions.go

```diff
@@ -0,0 +1,218 @@
+	"gopkg.in/yaml.v2"
+)
+
+type ContentYamlFile struct {
+	FunctionName string   `json:"function_name" yaml:"function_name"`
+	Description  string   `json:"description" yaml:"description"`
+	Tags         []string `json:"tags" yaml:"tags"`
+	Language     string   `json:"language" yaml:"language"`
+}
+
+type FunctionsResult struct {
+	FunctionName string    `json:"function_name"`
+	Description  string    `json:"description"`
+	Tags         []string  `json:"tags"`
+	Language     string    `json:"language"`
+	LastCommit   time.Time `json:"last_commit"`
+	Link         string    `json:"link"`
+	Repository   string    `json:"repository"`
+	Branch       string    `json:"branch"`
+}
+
+type FunctionsHandler struct{}
+
+func (fh FunctionsHandler) GetAllFunctions(c *gin.Context) {
+	user, err := getUserDetailsFromMiddleware(c)
+	if err != nil {
+		serv.Errorf("GetAllFunctions at getUserDetailsFromMiddleware: %v", err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	functions := []FunctionsResult{}
+
+	for k := range SourceCodeManagementFunctionsMap {
+		if tenantIntegrations, ok := IntegrationsConcurrentCache.Load(user.TenantName); !ok {
+			serv.Warnf("GetAllFunctions: %v", "the integration deos not exist")
+			c.AbortWithStatusJSON(SHOWABLE_ERROR_STATUS_CODE, gin.H{"message": err.Error()})
+			return
+		} else {
+			if integration, ok := tenantIntegrations[k].(models.Integration); ok {
+				connectedRepos, err := fh.GetConnectedSourceCodeRepos(integration)
+				if err != nil {
+					serv.Errorf("[tenant: %v][user: %v]GetAllFunctions at GetConnectedSourceCodeRepos: %v", user.TenantName, user.Username, err.Error())
+					continue
+				}
+				var repoContent interface{}
+				for _, connectedRepo := range connectedRepos {
+					connectedRepoRes := connectedRepo.(map[string]interface{})
+					repo := connectedRepoRes["repo_name"].(string)
+					owner := connectedRepoRes["repo_owner"].(string)
+					repoContent, err = fh.GetContentOfSelectedRepos(repo, owner, k, integration)
+					if err != nil {
+						serv.Errorf("[tenant: %v][user: %v]GetAllFunctions at GetContentOfSelectedRepos: %v", user.TenantName, user.Username, err.Error())
+						continue
+					}
+					functions, err = GetFunctionsDetails(user, k, integration, repoContent, connectedRepoRes, functions)
+					if err != nil {
+						serv.Errorf("[tenant: %v][user: %v]GetAllFunctions at GetFunctionsDetails: %v", user.TenantName, user.Username, err.Error())
+						continue
+					}
+				}
+			}
+		}
+	}
+	c.JSON(200, functions)
+}
+
+func validateYamlContent(yamlMap map[string]interface{}) error {
+	requiredFields := []string{"function_name", "description", "tags", "language"}
+	missingFields := make([]string, 0)
+	for _, field := range requiredFields {
+		if _, exists := yamlMap[field]; !exists {
+			missingFields = append(missingFields, field)
+		}
+	}
+
+	if len(missingFields) > 0 {
+		return fmt.Errorf("Missing fields: %v\n", missingFields)
+	}
+	return nil
+}
```

Comment:
It shouldn’t be a receiver function and it can start with lower case

Reply:
the function already with lower case and not receiver function